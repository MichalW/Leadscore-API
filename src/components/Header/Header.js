import PropTypes from 'prop-types';
import React from 'react';
import {Header, Icon} from 'semantic-ui-react';

const HeaderComponent = ({title, subtitle, icon}) => (
  <Header as="h2">
    <Icon name={icon} />
    <Header.Content>
      {title}

      {subtitle && (
        <Header.Subheader>{subtitle}</Header.Subheader>
      )}
    </Header.Content>

  </Header>
);

HeaderComponent.defaultProps = {
  subtitle: null,
  icon: null,
};

HeaderComponent.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  icon: PropTypes.string,
};

export default HeaderComponent;
