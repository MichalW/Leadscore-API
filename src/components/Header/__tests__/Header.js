import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';

import Header from '../Header';

it('should render Header with title', () => {
  const props = {
    title: 'Example title',
  };

  const header = shallow(<Header {...props} />);

  expect(toJson(header)).toMatchSnapshot();
});

it('should render Header with subtitle', () => {
  const props = {
    title: 'Example title',
    subtitle: 'Example subtitle',
  };

  const header = shallow(<Header {...props} />);

  expect(toJson(header)).toMatchSnapshot();
});

it('should render Header with icon', () => {
  const props = {
    title: 'Example title',
    subtitle: 'Example subtitle',
    icon: 'home',
  };

  const header = shallow(<Header {...props} />);

  expect(toJson(header)).toMatchSnapshot();
});
