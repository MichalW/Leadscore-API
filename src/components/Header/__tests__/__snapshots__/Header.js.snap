// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header with icon 1`] = `
<Header
  as="h2"
>
  <Icon
    as="i"
    name="home"
  />
  <HeaderContent>
    Example title
    <HeaderSubheader>
      Example subtitle
    </HeaderSubheader>
  </HeaderContent>
</Header>
`;

exports[`should render Header with subtitle 1`] = `
<Header
  as="h2"
>
  <Icon
    as="i"
    name={null}
  />
  <HeaderContent>
    Example title
    <HeaderSubheader>
      Example subtitle
    </HeaderSubheader>
  </HeaderContent>
</Header>
`;

exports[`should render Header with title 1`] = `
<Header
  as="h2"
>
  <Icon
    as="i"
    name={null}
  />
  <HeaderContent>
    Example title
  </HeaderContent>
</Header>
`;
